### PR TITLE
UCP/TAG: Optimize unexp eager offload handling

### PR DESCRIPTION
**Pull request by brminich**

### CODE REVIEW

File: src/ucp/core/ucp_request.inl [hunk 1]
```diff
@@ -500,11 +502,12 @@ ucp_recv_desc_init(ucp_worker_h worker, void *data, size_t length,
         }
 
         rdesc->flags = rdesc_flags;
-        memcpy(rdesc + 1, data, length);
+        data_hdr     = rdesc + 1;
+        memcpy((char*)data_hdr + data_offset, data, length);
```
Comment (hoopoepg) [inline src/ucp/core/ucp_request.inl]: minor: UCS_PTR_BYTE_OFFSET may be used

File: src/ucp/tag/eager_rcv.c [hunk 2]
```diff
@@ -13,6 +13,58 @@
...
+    ucp_worker_t *worker = arg;
+    ucp_request_t *req;
+    ucp_recv_desc_t *rdesc;
+    ucp_tag_t *rdesc_hdr;
+    ucs_status_t status;
+
+    req = ucp_tag_exp_search(&worker->tm, recv_tag);
+    if (req != NULL) {
+        ucp_eager_expected_handler(worker, req, data, length, recv_tag, flags);
+        req->recv.tag.info.length = length;
```
Comment (hoopoepg) [inline src/ucp/tag/eager_rcv.c]: perfectionism: alignment of ```=``` on this and next lines

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6633/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6634/ for details (Mellanox internal link).

File: src/ucp/tag/eager_rcv.c (hunk 2 above)
Comment (brminich) [reply src/ucp/tag/eager_rcv.c]: thought about this, but then the following lines would not fit to 80 sym limit

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6635/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6636/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6638/ for details (Mellanox internal link).

**Pull request merged**